### DISCUSSION TITLE
_markdown operator

### DISCUSSION THREAD

**Discussion by Radiergummi** [lowdefy/lowdefy#767] [Ideas and feature requests]
Sometimes I'd like to have the option to render Markdown into other text elements or properties of other blocks - for example the `extra` property of `TextInput` blocks, or `Paragraph` block content. 

<img width="585" alt="Bildschirmfoto 2021-08-20 um 14 50 05" src="https://user-images.githubusercontent.com/6115429/130235740-4dc4a6ed-f8bc-4eb0-a8df-5e5eb0fd5824.png">

On this screenshot, you can see the subtle style differences between Markdown (top text) and paragraph (bottom text). Here, I'd prefer the smaller font size of the paragraph block, but I require the inline formatting options of Markdown - which in turn lacks the ability to control the font size.  

My suggestion would be a `_markdown` operator that renders Markdown and passes on the result. That would make it possible to simply apply formatting in cases like the above.

**Comment by Gervwyk** [comment]
Hi @Radiergummi 

This is a cool idea. I'll have to investigate how markdown would affect since we are applying markdown specific styling using `react-markdown` see the [Markdown blocks](https://github.com/lowdefy/lowdefy/tree/main/packages/blocks/blocksMarkdown) I'd be happy to accept PRs to make the Markdown styling more aligned with the rest of antd. I've not spent enough time on this. 

The good news is in v3.20 you can render HTML into the label extra. [See this PR](https://github.com/lowdefy/lowdefy/pull/739). Which in this use case I think would be ideal? 

Alternatively, you should be able to control the font size inside a markdown element. I'll post an example in a few minutes.

**Reply by Radiergummi** [reply]
Rendering HTML is something I've been looking for the past few days, so yeah that will do! However having a `_markdown` operator on top of that would allow for using markdown seamlessly anywhere, which is kind of cool, I think. 

I've tried changing the font size like so:
```yaml
  - id: block_id
    type: Markdown
    properties:
      style:
        font_size: 12
      content: foo bar
```
that doesn't work because the HTML output looks like this:
```html
<div id="block_id" class="css-1qxtz39"> <!-- .css-1qxtz39 { font-size: 12px } -->
  <div class="markdown-body"> <!-- .markdown-body { font-size: 16px } -->
    <p>foo bar</p>
  </div>
</div>
```

**Reply by SamTolmay** [reply]
I was also thinking that having a `_markdown` operator that can convert into HTML would be great to use with the newly supported HTML in a lot of blocks. 

To change the font size in `Markdown` we usualy do:
```yaml
id: text
type: Markdown
style:
  '.markdown-body':
    fontSize: 14px
```
